src/solvers/solver_utils.hpp

```diff
@@ -0,0 +1,140 @@
+//========================================================================================
+// (C) (or copyright) 2021. Triad National Security, LLC. All rights reserved.
+//
+// This program was produced under U.S. Government contract 89233218CNA000001 for Los
+// Alamos National Laboratory (LANL), which is operated by Triad National Security, LLC
+// for the U.S. Department of Energy/National Nuclear Security Administration. All rights
+// in the program are reserved by Triad National Security, LLC, and the U.S. Department
+// of Energy/National Nuclear Security Administration. The Government is granted for
+// itself and others acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works, distribute copies to
+// the public, perform publicly and display publicly, and to permit others to do so.
+//========================================================================================
+#ifndef SOLVERS_SOLVER_UTILS_HPP_
+#define SOLVERS_SOLVER_UTILS_HPP_
+
+#include <string>
+#include <vector>
+
+#include "kokkos_abstraction.hpp"
+
+namespace parthenon {
+
+namespace solvers {
+
+struct SparseMatrixAccessor {
+  ParArray1D<int> ioff, joff, koff;
+  const int nstencil;
+  int ndiag;
+
+  SparseMatrixAccessor(const std::string &label, const int n,
+                       std::vector<std::vector<int>> off)
+      : ioff(label + "_ioff", n), joff(label + "_joff", n), koff(label + "_koff", n),
+        nstencil(n) {
+    assert(off.size() == 3);
+    assert(off[0].size() >= n);
+    assert(off[1].size() >= n);
+    assert(off[2].size() >= n);
```

Comment:
Can these `assert`s be `PARTHENON_REQUIRE_THROWS` instead? The Parthenon error checking macros give extra information about e.g. line and file numbers